src/toil/jobStores/aws/jobStore.py

```diff
@@ -1212,34 +1218,35 @@ def readFrom(self, readable):
                                     break
                                 info._update_checksum(hasher, buf)
                         except:
-                            with panic(log=log):
-                                for attempt in retry_s3():
-                                    with attempt:
-                                        upload.cancel_upload()
+                            client.abort_multipart_upload(Bucket=bucket_name,
+                                                          Key=compat_bytes(info.fileID), UploadId=uploadId)
                         else:
 
-                            while store._getBucketVersioning(store.filesBucket) != True:
-                                log.warning('Versioning does not appear to be enabled yet. Deferring multipart upload completion...')
+                            while not store._getBucketVersioning(store.filesBucket.name):
+                                log.warning('Versioning does not appear to be enabled yet. Deferring multipart upload '
+                                            'completion...')
                                 time.sleep(1)
-                                
+
                             # Save the checksum
                             info.checksum = info._finish_checksum(hasher)
 
-                            for attempt in retry_s3():
-                                with attempt:
-                                    log.debug('Attempting to complete upload...')
-                                    completed = upload.complete_upload()
-                                    log.debug('Completed upload object of type %s: %s', str(type(completed)), repr(completed))
-                                    info.version = completed.version_id
-                                    log.debug('Completed upload with version %s', str(info.version))
+                            log.debug('Attempting to complete upload...')
+                            completed = client.complete_multipart_upload(
```

Comment:
The docs recommend some error handling here: https://ibm.github.io/ibm-cos-sdk-python/reference/services/s3.html#S3.Client.complete_multipart_upload

Could we add some retries for 404, 500, 502, 503, and 504?